syntax = "proto3";
package criterion;

option java_package = "com.imageworks.spcue.grpc.criterion";
option java_multiple_files = true;

option go_package = "opencue_gateway/gen/go";

// -------- Primary Message Types --------]

message EqualsFloatSearchCriterion {
    float value = 1;
}

message EqualsIntegerSearchCriterion {
    int32 value = 1;
}

message GreaterThanFloatSearchCriterion {
    float value = 1;
}

message GreaterThanIntegerSearchCriterion {
    sint64 value = 1;
}

message InRangeFloatSearchCriterion {
    float min = 1;
    float max = 2;
}

message InRangeIntegerSearchCriterion {
    sint64 min = 1;
    sint64 max = 2;
}

message LessThanFloatSearchCriterion {
    float value = 1;
}

message LessThanIntegerSearchCriterion {
    sint64 value = 1;
}
